### PR TITLE
EI 203 apply amend apprenticeships selection

### PR DESCRIPTION
**Pull request by ruslanss**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives/InnerApi/Requests/PutIncentiveApplicationRequest.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+namespace SFA.DAS.EmployerIncentives.InnerApi.Requests
+{
+    public class PutIncentiveApplicationRequest : IPutApiRequest
+    {
+        public string BaseUrl { get; set; }
+        public string PutUrl => $"{BaseUrl}applications";
+        public object Data { get; set; }
+    }
+
+    public class UpdateIncentiveApplicationRequest
```
Comment (David-B-Read) [inline src/SFA.DAS.EmployerIncentives/InnerApi/Requests/PutIncentiveApplicationRequest.cs:13]: General convention is to have one class per file

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Steps/UpdateIncentiveApplicationSteps.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+            var request = new UpdateApplicationRequest
+            {
+                ApplicationId = _applicationId,
+                AccountId = _accountId,
+                AccountLegalEntityId = _accountLegalEntityId,
+                ApprenticeshipIds = _apprenticeshipIds
+            };
+
+            _response = await _context.OuterApiClient.PutAsync($"accounts/{_accountId}/applications", new StringContent(JsonSerializer.Serialize(request), Encoding.UTF8, "application/json"));
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Steps/UpdateIncentiveApplicationSteps.cs:75]: The response is never checked?

File: src/SFA.DAS.EmployerIncentives.Api/Controllers/ApplicationController.cs [hunk 3]
```diff
@@ -24,11 +25,20 @@ public ApplicationController(IMediator mediator, ILogger<ApplicationController>
...
 
-            return new CreatedResult($"/accounts/{request.AccountId}/applications/{request.ApplicationId}", null);
+            return new CreatedResult($"/accounts/{request.AccountId}/applications/{applicationId}", null);
+        }
+
+        [HttpPut]
+        [Route("/accounts/{accountId}/applications")]
+        public async Task<IActionResult> UpdateApplication(UpdateApplicationRequest request)
+        {
+            var applicationId = await _mediator.Send(new UpdateApplicationCommand(request.ApplicationId, request.AccountId, request.AccountLegalEntityId, request.ApprenticeshipIds));
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerIncentives.Api/Controllers/ApplicationController.cs:39]: What's the point of returning the applicationId?  Same question would go for CreateApplication but I didn't review that one.

File: src/SFA.DAS.EmployerIncentives.UnitTests/Application/EligibleApprenticeships/Commands/WhenHandlingUpdateApplicationCommand.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            long[] apprenticeshipIds,
+            Guid applicationId,
+            ApprenticeshipResponse[] apprenticeshipDetails,
+            [Frozen] Mock<IEmployerIncentivesService> employerIncentivesService,
+            [Frozen] Mock<ICommitmentsService> commitmentsService,
+            UpdateApplicationCommandHandler handler)
+        {
+
+            var command = new UpdateApplicationCommand(applicationId, accountId, accountLegalEntityId, apprenticeshipIds);
+            commitmentsService.Setup(x => x.GetApprenticeshipDetails(It.IsAny<long>(), It.IsAny<long[]>())).ReturnsAsync(apprenticeshipDetails);
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerIncentives.UnitTests/Application/EligibleApprenticeships/Commands/WhenHandlingUpdateApplicationCommand.cs:47]: Can the IsAnys be replaced with actual values?

File: src/SFA.DAS.EmployerIncentives.Api/Models/UpdateApplicationRequest.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace SFA.DAS.EmployerIncentives.Api.Models
+{
+    public class UpdateApplicationRequest
+    {
+        public Guid ApplicationId { get; set; }
+        public long AccountId { get; set; }
+        public long AccountLegalEntityId { get; set; }
```
Comment (Pauljgraham) [inline src/SFA.DAS.EmployerIncentives.Api/Models/UpdateApplicationRequest.cs:9]: I thought we weren't sending in the AccountLegalEntity as it already exists on the application and can not be changed

Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Api/Models/UpdateApplicationRequest.cs:9]: good spot!

**Pull request merged**